### ISSUE TITLE
Release plugin version 1.0.0

### ISSUE THREAD

**Issue by EMH333** [EMH333/esbuild-svelte#204]
I am planning to release v1.0.0 of this plugin once Svelte 5 is released. esbuild development seems to have mostly stabilized so I'm feeling okay moving forward without esbuild reaching v1.

Some changes that will likely happen:
- [ ] Minimum esbuild version set to `v0.17.0` to simplify version support
- [ ] Minimum Svelte version set to `v4` (or `v5` depending on how breaking the changes are) to simplify version support
- [ ] Removal of `overzealous` cache option
- [ ] #188 
- [ ] Stretch: #24

**Comment by gka** [comment]
Would be useful if there was a pre-release of v1.0.0 at some point _before_ Svelte 5 gets released.

**Comment by EMH333** [comment]
Agreed, I intend to start a svelte-v5 branch in the near future. I also haven't been closely following Svelte 5 development, is there a timeline to release yet?

**Comment by gka** [comment]
Nothing official yet, other than "early 2024", which at this point probably means Q2. The closest to a timeline is probably [this myserious tweet](https://twitter.com/Rich_Harris/status/1758543656170975596), though 😆

**Comment by xelius** [comment]
Svelte 5 reached RC now 🎉 and I would be really happy to see this great package support it. 

 (https://www.youtube.com/watch?v=gkJ09joGBZ4&t=9254s)

**Comment by EMH333** [comment]
I've scoped out the changes for Svelte 5 and they seem fairly reasonable for simple support alongside Svelte 3 and 4. I'll look to get a non-breaking release supporting Svelte 5 out in the next few days

**Comment by iskrisis** [comment]
I've unsucessfuly tried to use version that's in the `svelte-5` branch. Could you maybe give us some pointers how you test it or make pre-release? I am currently writing small Svelte 5 app using vite but always prefered your plugin. Seems like there would be few other testers. 🙏 🙏

**Comment by EMH333** [comment]
@iskrisis Please try the `svelte-5-compat` branch used by #230 which is more geared towards graceful compatibility between Svelte 3, 4 and 5 instead of removing support for the lower version. Building the dist/ folder with `npm run build` should be all that is required to output a working plugin for any Svelte version >= 3. I'll note that in compatibility testing, the `svelte-5-compat` branch [is able to successfully compile files with Svelte 5](https://github.com/EMH333/esbuild-svelte/actions/runs/9121463901/job/25080654199?pr=230#step:7:208).

If I can get a few :rocket: reactions or comments on that PR with folks who have been able to test it successfully, I'll go ahead and release it as `0.8.1`. Otherwise, to avoid the risk breaking preexisting projects, I'll probably aim for a `0.9.0` release before `1.0.0` to coincide with Svelte 5.

**Let's please move conversation about Svelte 5 compatibility to #230 since that will make it easier to track specific breakage/issues**

**Comment by EMH333** [comment]
Version `v0.8.1` has been released which relaxes the Svelte peer dependency version lock to include Svelte 5. Please give it a try!

**Comment by EMH333** [comment]
I've opened #234 to track overall Svelte 5 compatibility and #235 to track compatibility with the prereleases. There are going to be several conversations happening at once and I want to keep it easy to follow what issues relate to which versions

@tgf9 I'm moving the resolution of your report (and the corresponding rabbit hole) to #235
